src/python/ensembl/io/genomio/genome_metadata/dump_genome_stats.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generates a JSON file representing various stats for the assembly and annotation from a core db.
+"""
+
+import json
+from pathlib import Path
+from typing import Any, Dict
+
+import argschema
+from sqlalchemy import select, func
+from sqlalchemy.engine import URL
+from sqlalchemy.orm import Session
+
+from ensembl.database import DBConnection
+from ensembl.core.models import SeqRegionAttrib, AttribType, Gene, Transcript
+
+
+class StatsGenerator:
+    """Interface to extract stats from a core database."""
+
+    def __init__(self, session: Session) -> None:
+        self.session = session
+
+    def get_assembly_stats(self) -> Dict[str, Any]:
+        """Returns a dict of stats about the assembly."""
+        stats = {
+            "coord_system": self.get_attrib_counts("coord_system_tag"),
+            "locations": self.get_attrib_counts("sequence_location"),
+            "codon_table": self.get_attrib_counts("codon_table"),
+        }
+
+        # Special: rename supercontigs to scaffolds for homogeneity
+        stats = self._fix_scaffolds(stats)
+        return stats
+
+    def _fix_scaffolds(self, stats: Dict[str, Any]):
+        coords = stats.get("coord_system", {})
+        if "supercontig" in coords:
+            if "scaffold" in coords:
+                pass
+            else:
+                coords["scaffold"] = coords["supercontig"]
+                del coords["supercontig"]
```

Comment:
Unless you plan to add code in the near future, it is advisable to avoid using `pass`:
```suggestion
            if "scaffold" not in coords:
                coords["scaffold"] = coords["supercontig"]
                del coords["supercontig"]
```